woocommerce/woocommerce-ios
# Show stats v4 by default if possible

# Why
As part of https://github.com/woocommerce/woocommerce-ios/issues/1946 we will be updating the stats banner to let the user know that the v3 stats will be deprecated soon.

This PR changes the behavior to the following:
- If stats v4 are available, we default to show them and show no banner
- If stats v4 are not available, we show stats v3 dashboards and show a collapsable banner indicating the user that the v3 stats will be deprecated soon.

Changes in following PRs will include:
- Dismiss and remind me later actions
- Empty state screen if the user hasn't updated it's Woo version and September 1st has arrived.

# How

First, we have reduced the `StatsVersionState` enum to only have one case(`initial`) because we don't care anymore about transitioning between v4 or v3 states. In the future we aim to remove this enum and only store it's associated property `statsVersion`. While doing that we have also trimmed a lot of tests that weren't needed anymore. 

Second, we have added a new banner using `TopBannerView` to indicate the user that they are using and old and soon to be deprecated version of the stats. With this we have deleted the code associated with `V3ToV4` and `V4ToV3` banners.

# Screenshots

V3 | V4
--- | ---
<img width="425" alt="v3" src="https://user-images.githubusercontent.com/562080/85430425-b7d85380-b545-11ea-8153-87dbcc063a41.png">|<img width="442" alt="v4" src="https://user-images.githubusercontent.com/562080/85430416-b575f980-b545-11ea-9209-29a8e25a12a1.png">


# Testing steps
- Run the app using a store that doesn't have v4 stats available and see that the correct banner(indicating the user to upgrade) is being shown.
- Run the app using a store that has v4 stats available and see that the new stats dashboard is being shown and not banner is present.


# Questions
- The android designs have a black tone for the icon on the banner, other banners on the iOS app have a gray-ish tone, should we keep the gray tone for this banner too @Garance91540 ?


Update release notes:
